src/python_testing/basic_composition_support.py

```diff
@@ -98,12 +98,15 @@ def ConvertValue(value) -> Any:
 
 
 class BasicCompositionTests:
-    async def connect_over_pase(self, dev_ctrl):
-        asserts.assert_true(self.matter_test_config.qr_code_content == [] or self.matter_test_config.manual_code == [],
-                            "Cannot have both QR and manual code specified")
-        setupCode = self.matter_test_config.qr_code_content + self.matter_test_config.manual_code
-        asserts.assert_equal(len(setupCode), 1, "Require one of either --qr-code or --manual-code.")
-        await dev_ctrl.FindOrEstablishPASESession(setupCode[0], self.dut_node_id)
+    def get_code(self, dev_ctrl):
+        created_codes = []
+        for idx, d in enumerate(self.matter_test_config.discriminators):
+            created_codes.append(dev_ctrl.CreateManualCode(d, self.matter_test_config.setup_passcodes[idx]))
+
+        setup_codes = self.matter_test_config.qr_code_content + self.matter_test_config.manual_code + created_codes
```

Comment:
`manual_code` would have to be a list, no?

If a manual code is actually passed, and it's not a list, then it's possible adding list + str + list will fail.

Reply:
yup - it's a list. See https://github.com/project-chip/connectedhomeip/blob/master/src/python_testing/matter_testing_support.py#L640. Was changed when we did the DA test that require multiple devices. 

Manual code passing in tested in one of the CI runs.